cmd/snap-seccomp/main_test.go

```diff
@@ -350,8 +351,20 @@ func (s *snapSeccompSuite) TestUnrestricted(c *C) {
 //    {"read >=2", "read;native;1", main.SeccompRetKill},
 //    {"read >=2", "read;native;0", main.SeccompRetKill},
 func (s *snapSeccompSuite) TestCompile(c *C) {
-	// The 'shadow' group is different in different distributions
-	shadowGid, err := osutil.FindGid("shadow")
+	// The 'shadow' group is different in different distributions, if group
+	// does not exist, instead of guessing gid based on group name, look at
+	// /etc/shadow and use the gid that owns it
+	var shadowGid uint64
+	if gid, err := osutil.FindGid("shadow"); err != nil {
+		var stat syscall.Stat_t
```

Comment:
Why not just `os.Stat`?

Reply:
`os.Stat` does not carry any group information I'm afraid.